### PR TITLE
New loadEmojis hook, to customize emojis at runtime.

### PR DESCRIPTION
**Pull request by JohnXLivingston**

### CODE REVIEW

File: src/headless/plugins/emoji/plugin.js [hunk 1]
```diff
@@ -75,8 +75,30 @@ converse.plugins.add('converse-emoji', {
                 async initialize () {
                     if (!converse.emojis.initialized) {
                         converse.emojis.initialized = true;
+
                         const module = await import(/*webpackChunkName: "emojis" */ './emoji.json');
-                        const json = (converse.emojis.json = module.default);
+                        /**
+                         * *Hook* which allows plugins to modify emojis definition.
+                         *
+                         * Note: This hook is only fired one time, when ConverseJS initiliazed.
```
Comment (jcbrand) [inline src/headless/plugins/emoji/plugin.js:83]: Typo: initiliazed

Also, we use just `Converse` in the docs.

File: src/headless/plugins/emoji/plugin.js [hunk 2]
```diff
@@ -75,8 +75,30 @@ converse.plugins.add('converse-emoji', {
...
+                         *
+                         * @event _converse#loadEmojis
+                         * @param json
+                         *      See src/headless/emojis.json for more information about the content of this parameter.
```
Comment (jcbrand) [inline src/headless/plugins/emoji/plugin.js:87]: Would be nice if you used the JSDoc `@link` tag to make a hyperlink.

File: src/headless/plugins/emoji/plugin.js (hunk 1 above)
Comment (JohnXLivingston) [reply src/headless/plugins/emoji/plugin.js]: fixed.

File: src/headless/plugins/emoji/plugin.js (hunk 2 above)
Comment (JohnXLivingston) [reply src/headless/plugins/emoji/plugin.js]: Done.

File: src/headless/plugins/emoji/plugin.js [hunk 3]
```diff
@@ -75,8 +75,35 @@
...
+                         *      json.custom[":my_emoji"] = {
+                         *          "sn":":my_emoji:","url":"https://example.com/my_emoji.png","c":"custom"
+                         *      };
+                         *      delete json.custom[":converse:"];
+                         *      return json;
+                         *  });
+                         */
+                        const json = (
+                            await api.hook('loadEmojis', {}, module.default)
+                        )
```
[the remark this answers is not in the document]
Comment (jcbrand) [reply src/headless/plugins/emoji/plugin.js:103]: @JohnXLivingston :point_up:

**Pull request merged**